Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -142,91 +143,58 @@ def partitionByScript(
         """Split a potentially mixed-script pair into pairs that make sense based
         on the dominant script, and yield each combination with its dominant script."""
 
-        # First, partition the pair by their assigned scripts
-        allFirstScripts: dict[tuple[str, ...], list[str]] = {}
-        allSecondScripts: dict[tuple[str, ...], list[str]] = {}
+        # First, partition the pair by their assigned scripts. Glyphs can have
+        # multiple scripts assigned to them (legitimately, e.g. U+0951
+        # DEVANAGARI STRESS SIGN UDATTA, or for random reasons like having both
+        # `sub h by h.sc` and `sub Etaprosgegrammeni by h.sc;`). We duplicate
+        # the glyph name into each script bucket because each script gets its
+        # own lookup and group membership is exclusive per lookup.
+        side1Scripts: dict[str, list[str]] = {}
+        side2Scripts: dict[str, list[str]] = {}
         for glyph in self.firstGlyphs:
-            if glyph not in glyphScripts:
-                glyphScripts[glyph] = {COMMON_SCRIPT}
-            allFirstScripts.setdefault(tuple(glyphScripts[glyph]), []).append(glyph)
+            for script in glyphScripts.get(glyph, COMMON_SCRIPTS_SET):
+                side1Scripts.setdefault(script, []).append(glyph)
         for glyph in self.secondGlyphs:
-            if glyph not in glyphScripts:
-                glyphScripts[glyph] = {COMMON_SCRIPT}
-            allSecondScripts.setdefault(tuple(glyphScripts[glyph]), []).append(glyph)
+            for script in glyphScripts.get(glyph, COMMON_SCRIPTS_SET):
+                side2Scripts.setdefault(script, []).append(glyph)
 
```

Comment:
Aaargh. For pairs like `pos @kern1.C tbar ...`, this code turns the single glyph on either side into a list with a single glyph, which is why the `KerningPair` 's init is turning lists with 1 occupant back into glyphs rather than lists. So just removing that logic creates new glyph groups that fail sanity checking.